pages/create/index.js

```diff
@@ -0,0 +1,246 @@
+
+  @media screen and (min-width: 1024px) {
+    padding: 5% 0;
+  }
+`;
+
+const StyledForm = styled.form`
+  display: flex;
+  flex-direction: column;
+  width: 80vw;
+  max-width: 800px;
+  height: 30vh;
+  min-height: 250px;
+  border: 3px solid black;
+  padding: 10px;
+  background: var(--secondary-color);
+  border-radius: 20px;
+  box-shadow: 5px 5px 5px 5px black;
+  font-size: 1rem;
+  font-weight: 800;
+
+  @media screen and (min-width: 375px) {
+    font-size: 0.8rem;
+  }
+`;
+
+const StyledConfirmButtonContainer = styled.div`
+  display: flex;
+`;
+
+const StyledIndicatorContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: space-evenly;
+  width: 70%;
+  max-width: 600px;
+  height: 30%;
+  min-height: 200px;
+  border: 2px solid black;
+  background: var(--secondary-color);
+  border-radius: 20px;
+  box-shadow: 5px 5px 5px 5px black;
+
+  @media screen and (min-width: 375px) {
+    font-size: 0.8rem;
+  }
+`;
+
+const StyledCharacteristicsContainer = styled.div`
+  display: flex;
+  gap: 10px;
+
+  select {
+    width: 22vw;
+    max-width: 240px;
+    border-radius: 5px;
+
+    @media screen and (min-width: 375px) {
+      font-size: 0.7rem;
+    }
+  }
+`;
+
+const StyledFormArticle = styled.article`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  gap: 10px;
+  height: 100%;
+  width: 100%;
+
+  label,
+  input {
+    border: 2px solid black;
+    padding: 5px;
+    width: 70%;
+    border-radius: 10px;
+  }
+
+  label {
+    width: 20%;
+    border: none;
+  }
+`;
+
+export default function CreatePetPage({ onCreatePet }) {
+  const [currentPet, setCurrentPet] = useState(0);
+  const formRef = useRef(null);
+
+  const [characteristics, setCharacteristics] = useState({
+    characteristic1: "",
+    characteristic2: "",
+  });
+
+  function handlePreviousPet() {
+    const prevPetId = currentPet > 0 ? currentPet - 1 : animalList.length - 1;
+    setCurrentPet(prevPetId);
+  }
+
+  function handleNextPet() {
+    const nextPetId = (currentPet + 1) % animalList.length;
+    setCurrentPet(nextPetId);
+  }
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+    const petData = {
+      ...data,
+      type: animalList[currentPet].type,
+      picture: animalList[currentPet].icon,
+    };
+    event.target.reset();
+    setCharacteristics({
+      characteristic1: "",
+      characteristic2: "",
+    });
+    onCreatePet(petData);
+  }
+
+  function handleExternalButtonClick() {
+    if (formRef.current) {
+      formRef.current.requestSubmit();
+    }
+  }
+
+  return (
+    <StyledCreatePage>
+      <PetSelection
+        onPreviousPet={handlePreviousPet}
+        onNextPet={handleNextPet}
+        animalList={animalList}
+        currentPet={currentPet}
+        name="type"
+      />
+      <StyledForm ref={formRef} onSubmit={handleSubmit}>
```

Comment:
I created a new component but now the code has become much more complicated because I had to implement ref and forwardRef. It will take me some time to figure out how that works 😅

Reply:
You can create a PetForm component. That component can contain all the stuff you need there. Including the useRef, the button, etc. 
You only have to be aware not to include unnecessary  elements inside the `<form></form>` element itself.